azurerm/internal/services/apimanagement/api_management_api_diagnostic_resource.go

```diff
@@ -182,29 +182,16 @@ func resourceApiManagementApiDiagnosticCreateUpdate(d *schema.ResourceData, meta
 	}
 
 	if verbosity, ok := d.GetOk("verbosity"); ok {
-		switch verbosity.(string) {
-		case string(apimanagement.Verbose):
-			parameters.Verbosity = apimanagement.Verbose
-		case string(apimanagement.Information):
-			parameters.Verbosity = apimanagement.Information
-		case string(apimanagement.Error):
-			parameters.Verbosity = apimanagement.Error
-		}
+		parameters.Verbosity = apimanagement.Verbosity(verbosity.(string))
 	}
 
-	if logClientIP, ok := d.GetOk("log_client_ip"); ok {
+	logClientIP, _ := d.GetOk("log_client_ip")
+	if logClientIP != nil {
 		parameters.LogClientIP = utils.Bool(logClientIP.(bool))
 	}
```

Comment:
Hi @slynickel, many thanks for the PR.

Because this attribute is optional+computed, it would be best to use `GetOkExists()` which tells you if the attribute is configured, but bypasses the zero value check so you can detect an explicit false value.

```suggestion
	if logClientIP, exists := d.GetOkExists("log_client_ip"); exists {
		parameters.LogClientIP = utils.Bool(logClientIP.(bool))
	}
```

Reply:
@manicminer Isn't that function being deprecated though? [ResourceData.GetOkayExist()](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/schema#ResourceData.GetOkExists)

>GetOkExists can check if TypeBool attributes that are Optional with no Default value have been set.
>
>Deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK

My light review of the terraform-plugin-sdk is that some sections are pretty old, though the deprecation warning is within the last year.

Reply:
@slynickel I checked with the team and although it's marked as deprecated in the SDK, in a context like this there's little alternative and it's ok to use it.

Reply:
Updated.

Make sense, thanks for context.

Reply:
@manicminer I'm assuming the failed golint check as a result of GetOkExist can be ignored?

Reply:
@slynickel we can add a nolint comment to suppress that, i've pushed that whilst I run the acceptance tests